graylog2-web-interface/src/components/common/Icon.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import 'font-awesome/css/font-awesome.css';
+
+/**
+ * Component that renders an icon or glyph.
+ * Uses Font Awesome 4.7 : https://fontawesome.com/v4.7.0/icons/
+ *
+ * No need to pass `fa` or `fa-` prefixes, just the name of the icon
+ */
+
+const Icon = React.forwardRef(({
+  className,
+  flip,
+  fixedWidth,
+  inverse,
+  name,
+  pulse,
+  rotate,
+  size,
+  spin,
+  ...props
+}, ref) => {
+  const cleanIconName = name.replace(/^fa-/, ''); // remove "fa-" prefix if it exists
+
+  return (
+    <i className={`
+         fa
+         fa-${cleanIconName}
+         ${flip && `fa-flip-${flip}`}
```

Comment:
All of these optional parameters do generate unwanted values (`undefined`/`false`) if they are not set because of these conditionals. Instead, we should build the class names string conditionally and only add classes if the properties are true, instead of adding the result of the conditionals.

You can see the effect of the current logic pretty well in the generated snapshots.